codegen/syntax/TypeConstraint.hack

```diff
@@ -136,22 +136,22 @@ final class TypeConstraint extends Node {
   }
 
   /**
-   * @return ClassnameTypeSpecifier | ClosureTypeSpecifier |
+   * @return ClassnameTypeSpecifier | ClosureTypeSpecifier | Contexts |
    * GenericTypeSpecifier | LikeTypeSpecifier | NullableTypeSpecifier |
    * ShapeTypeSpecifier | SimpleTypeSpecifier | TypeConstant |
    * VarrayTypeSpecifier | VectorTypeSpecifier
    */
-  public function getType(): ITypeSpecifier {
-    return TypeAssert\instance_of(ITypeSpecifier::class, $this->_type);
+  public function getType(): Node {
```

Comment:
Should we add `Contexts` to `ITypeSpecifier` too?

My plan was to create new interfaces for these (which I guess would mean also adding a new `wrap_IFoo` function for each) since neither `Context` nor `VariableToken` is really a type specifier, but now I'm not sure if that's worth the extra complexity.

Reply:
`HackAstNodeTypeConstraint::getType()` returns an `IHackAstNodeSpecifierOrContexts` which is basically `ITypeSpecifier|Context` - I think we probably want to do something similar here :(

Reply:
Sounds good, that's pretty much exactly what I independently arrived at (except my interface name was `ITypeConstraintType`).